forum/qa-plugin/report-reason/frontend/dist/script.js

```diff
@@ -0,0 +1 @@
uest-feedback">\n\t\t\t\t<div id="reportReasonRequestInfo" class="report-reason-popup__request-feedback-info"></div>\n\t\t\t\t<button id="closeReportReasonRequestFeedback"\n\t\t\t\t\tclass="report-reason-popup__button report-reason-popup__button--close"\n\t\t\t\t\ttype="button">'.concat(P.CLOSE,"</button>\n\t\t\t</div>\n\t\t")}},{key:"showReportReasonPopup",value:function(){this.formInvalidityListenerAPI.attach(),this.reportReasonPopupDOMWrapper.classList.add("display-block"),this.getFormDOM().elements.reportReason[0].focus()}},{key:"hideReportReasonPopup",value:function(){this.formInvalidityListenerAPI.detach(),this.reportReasonPopupDOMReferences.reportReasonRequestFeedback.classList.remove("display-block"),this.reportReasonPopupDOMWrapper.classList.remove("display-block"),this.reportReasonPopupDOMReferences.customReportReason.parentNode.classList.add("display-none"),this.reportReasonPopupDOMReferences.reportReasonPopup.classList.remove("display-none"),this.getReportReasonValidationErrorDOM().classList.remove("display-block");var t=this.getFormDOM();t.reset(),t.elements.customReportReason.required=!1,t.elements.sendReportReason.disabled=!1,this.resetCustomReportReasonCharCounter(),this.getReportReasonValidationErrorDOM().innerHTML=M.GENERIC_ERROR}},{key:"onAjaxSuccess",value:function(t){var e=t.newFlags,r=t.formData,o=this.getFlagButtonDOM(),n=this.updateCurrentPostFlags(e,r)?P.REPORT_SENT:M.GENERIC_ERROR;n===P.REPORT_SENT&&this.swapFlagBtn(o,O({postType:r.postType,questionId:r.questionId,postId:r.postId,parentId:this.getPostParentId(r.postType,o)})),this.showFeedbackPopup(n)}},{key:"showFeedbackPopup",value:function(t){this.reportReasonPopupDOMReferences.reportReasonRequestInfo.innerHTML=t,this.reportReasonPopupDOMReferences.reportReasonPopup.classList.add("display-none"),this.reportReasonPopupDOMReferences.reportReasonRequestFeedback.classList.add("report-reason-popup","display-block"),this.reportReasonPopupDOMReferences.closeReportReasonRequestFeedback.focus()}},{key:"initSuccessPopupCloseBtn",value:function(){this.reportReasonPopupDOMReferences.closeReportReasonRequestFeedback.addEventListener("click",this.hideReportReasonPopup.bind(this))}}]),t}(),E=function(){var t=new p,e=new y,r=new D({getFlagButtonDOM:t.getFlagButtonDOM.bind(t),formInvalidityListenerAPI:e.formInvalidityListenerAPI,getFormDOM:e.getFormDOM.bind(e),getPostParentId:t.getPostParentId.bind(t),swapFlagBtn:t.swapFlagBtn.bind(t),updateCurrentPostFlags:t.updateCurrentPostFlags.bind(t),getReportReasonValidationErrorDOM:e.getReportReasonValidationErrorDOM.bind(e),resetCustomReportReasonCharCounter:e.resetCustomReportReasonCharCounter.bind(e)});return e.initButtons({collectForumPostMetaData:t.collectForumPostMetaData.bind(t),hideReportReasonPopup:r.hideReportReasonPopup.bind(r),onAjaxSuccess:r.onAjaxSuccess.bind(r),showFeedbackPopup:r.showFeedbackPopup.bind(r)}),t.onClick.bind(t,r.showReportReasonPopup.bind(r))}();document.querySelector(".qa-main").addEventListener("click",E,!0)}]);
```

Comment:
Nie siedzę w technologiach webowych, ale zajrzałem z ciekawości i mam małe pytanie z kategorii naiwnych. :)
Czemu to jest takie paskudne? :D W sensie wygląda jak wygenerowane automatycznie w taki sposób żeby miało jak najmniej znaków. Czy to jest rzeczywiście działanie jakiegoś generatora?

Reply:
To jest zminifikowany kod, wygenerowany przez Webpacka i tak, to jest celowe działanie, żeby po prostu zmniejszyć rozmiar pliku.

Reply:
Ok, dzięki za potwierdzenie!